cli/tests/integrations/test_marathon.py

```diff
@@ -767,9 +776,20 @@ def _list_versions(app_id, expected_count, max_count=None):
 
     assert returncode == 0
     assert isinstance(result, list)
-    assert len(result) == expected_count
     assert stderr == b''
 
+    # Marathon persists app versions indefinitely by ID, so there may be extras
+    assert len(result) >= expected_min_count
+
+    if max_count is not None:
+        assert len(result) <= max_count
+
+
+def _assert_matches_with_groups(pattern, text, groups):
+    match = re.fullmatch(pattern, text, flags=re.DOTALL)
+    assert match
+    assert match.groups() == groups
```

Comment:
Yeah, that's less code, although it will be more work to change if `ZERO_INSTANCE_APP_ID` ever changes (unlikely).